src/arrayMethodReduce.js

```diff
@@ -3,10 +3,40 @@
 /**
  * Implement method Reduce
  */
+
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const startValue = arguments.length <= 1
+      ? 1
+      : 0;
+
+    let previousValue = arguments.length <= 1
+      ? this[0]
+      : initialValue;
+
+    for (let i = startValue; i < this.length; i++) {
```

Comment:
```suggestion
    for (let i = startIndex; i < this.length; i++) {
```
it's index, not value

Reply:
Changed code, check it out